account_bank_statement_import/models/account_bank_statement_import.py

```diff
@@ -325,12 +337,19 @@ def _create_bank_account(
     def _complete_statement(self, stmt_vals, journal_id, account_number):
         """Complete statement from information passed."""
         stmt_vals['journal_id'] = journal_id
+        statement_bank = self._get_bank(account_number)
         for line_vals in stmt_vals['transactions']:
-            unique_import_id = line_vals.get('unique_import_id', False)
+            unique_import_id = (
+                statement_bank.enforce_unique_import_lines and
+                'data' in line_vals and line_vals['data'] and
+                hashlib.md5(line_vals['data']).hexdigest() or
```

Comment:
I think not all the values should be taken into account for the hash (I don't know if right now for example `'statement_id'` is a key here, but with that, lines will never be the same). Only operation date, amount and concept should be taken into account.

Reply:
What about this comment?

Reply:
I answered that in the comment below this one. So statement_id is not part of the line data, and we need to take the maximum amount of information in the line into account, in order to prevent false positives on the test for duplication.

Reply:
Sorry, I didn't see it, but I insist as in upper versions there's no need of this mechanism. Why here it is needed?

Reply:
@pedrobaeza what do you mean that it is needed in later versions? Isn't this just a feature that can be forward ported?

Reply:
Just did a test with account_bank_statement_camt_oca in Odoo 11, and if I import the same statement twice the duplicate entries are not detected as such so I end up with a corruption of my bank ledger.

Reply:
Our customer does have conflicts, as any customer would have, if they download statements multiple times a day. This is done as their sales orders are only confirmed for delivery when proof of pay is received. So the same lines will be in the statements downloaded in the morning as for a few our later.

Reply:
But then the patch should be done on *_camt module, not on account_bank_statement_import, and more because in >=v9, we don't have control anymore over that module as it's integrated in core.